src/geo/transform.js

```diff
@@ -1417,15 +1417,56 @@ class Transform {
         return new LngLatBounds(new LngLat(west, south), new LngLat(east, north));
     }
 
+    // Gets minimum bounding lngLats for non-Mercator and non-globe projections.
+    _getBoundsProjection(): LngLatBounds {
+        assert(!this.projection.supportsTerrain, "This function doesn't account for terrain");
+        assert(!this.projection.supportsWorldCopies, "Projections that support world copies are rectangular and should the simpler _getProjection");
+        const {top, left} = this._edgeInsets;
+        const bottom = this.height - this._edgeInsets.bottom;
+        const right = this.width - this._edgeInsets.right;
+
+        const res = 20; // Number of points to sample along each edge
+        const w = right - left;
+        const h = bottom - top;
+
+        let west = 180;
+        let east = -180;
+        let south = 90;
+        let north = -90;
+
+        const checkExtreme = (x: number, y: number) => {
+            const {lng, lat} = this.pointLocation(new Point(x, y));
+            west = Math.min(west, lng);
+            east = Math.max(east, lng);
+            south = Math.min(south, lat);
+            north = Math.max(north, lat);
+        };
+
+        for (let i = 0; i <= res; i++) {
+            checkExtreme(left + w * i / res, top);
+            checkExtreme(right, top + h * i / res);
+            checkExtreme(right - w * i / res, bottom);
+            checkExtreme(left, bottom - h * i / res);
+        }
+
+        return new LngLatBounds(new LngLat(west, south), new LngLat(east, north));
+    }
+
     _getBounds(min: number, max: number): LngLatBounds {
         if (this.projection.name === 'globe') {
             return this._getGlobeBounds();
         }
```

Comment:
Can we move that up a level in the call stack so all the projection checks and dispatch to their related `_getBounds*` and the fork is in the same place? e.g., something like:

```js
    getBounds(): LngLatBounds {
        if (projection.name === "mercator" || projection.name === "equirectangular") {
             if (projection.supportsTerrain && this._terrainEnabled()) {
                 return this._getBoundsRectangular3D();
             }
             return this._getBoundsRectangular(...);
        }
        if (projection.name === "globe") {
            return this._getBoundsGlobe();
        }
        return this._getBoundsNonRectangular();
    }
```

Reply:
Good call